kpi/filters.py

```diff
@@ -133,15 +133,13 @@ def _get_queryset_for_collection_statuses(self, request, queryset):
 
         elif status == ASSET_STATUS_PUBLIC:
             self._return_queryset = True
-            # ToDo Review for optimization
             public = self._get_public(queryset)
             subscribed = self._get_subscribed(public, user)
             return subscribed
 
         elif status == ASSET_STATUS_DISCOVERABLE:
             self._return_queryset = True
-            # ToDo Review for optimization
-            discoverable = self._get_discoverable(self._get_public(queryset))
+            discoverable = self._get_discoverable(queryset)
```

Comment:
IMHO: `discover_asset` implies `view_asset`. No need to add a check on `view_asset`.

Reply:
yes, now that the backend enforces the implication, i agree